### PR TITLE
Only fetch updated notifications since the last sync

### PR DESCRIPTION
**Pull request by pixeltrix**
If you have a lot of notifications then downloading them all every time is going to be painful and probably blow the rate limit on the API.

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -13,7 +13,56 @@ class Notification < ApplicationRecord
...
+            subject_type: notification.subject.type,
+            reason: notification.reason,
+            unread: notification.unread,
+            updated_at: notification.updated_at,
+            last_read_at: notification.last_read_at,
+            url: notification.url
+          }
+
+          n.update(attrs)
+        rescue ActiveRecord::RecordNotUnique => e
```
Comment (tarebyte) [inline app/models/notification.rb]: Nice 👍

**Comment by pixeltrix** [comment]
Done 👍

**Comment by pixeltrix** [comment]
Hmm, don't merge it yet - just blew up locally

**Comment by andrew** [comment]
👍 We should probably also add a limit on the initial sync of perhaps just notifications from the past week/month to avoid download potentially huge amounts of old notifications.

**Comment by pixeltrix** [comment]
Okay, enhanced the test to actually create notifications and we only download a months worth on first sync

File: test/models/notification_test.rb [hunk 2]
```diff
@@ -1,7 +1,33 @@
 require 'test_helper'
 
 class NotificationTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  setup do
+    notifications_url = %r{https://api.github.com/notifications}
+
+    body     = File.read(Rails.root.join('test', 'fixtures', 'files', 'notifications.json'))
```
Comment (pixeltrix) [inline test/models/notification_test.rb]: Possibly - I should know about that shouldn't I? 😊

Comment (pixeltrix) [inline test/models/notification_test.rb]: Done

**Comment by andrew** [comment]
🚢

**Pull request merged**